### PR TITLE
vmdisplay-server: general clean-ups

### PR DESCRIPTION
**Pull request by downor**
- removed uncessary member functions and variables
- cleaned up routine to use less # of variables
- use "C++ style" comment style in C++ source and headers

Signed-off-by: root <root@dongwonk-Z87X-UD5H.fm.intel.com>

### CODE REVIEW

File: clients/vmdisplay/vm-shared.h [hunk 1]
```diff
@@ -37,16 +37,14 @@
 
 #define SURFACE_NAME_LENGTH     64
 #define BIT(a)                  (1<<a)
-/*
- * If a buffer has been updated by a VM client app, the status field will have
- * this UPDATED bit set. Otherwise, it would have this bit cleared
- */
+
```
Comment (satyeshwar) [inline clients/vmdisplay/vm-shared.h:40]: In general, we don't allow C++ style comments anywhere in IAS code. So I cannot accept this change unless you revert back to C style comments again.

**Review by satyeshwar — CHANGES_REQUESTED** [review]

File: clients/vmdisplay/vm-shared.h (hunk 1 above)
Comment (downor) [reply clients/vmdisplay/vm-shared.h:40]: hmm I saw // was mostly used in .cpp in example folder, such as many .cpp under clients. But if it's your policy, then I will remove those changes.

**Comment by satyeshwar** [comment]
Yeah Arun may have done that accidentally, but as a rule we only allow C style comments.

**Comment by satyeshwar** [comment]
Closing this pull request. Please submit another one without the C++ style comments.

**Pull request closed without merging**